contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8148,42 +7888,45 @@ rewrite_geospatial_col_ref_query_helper(T ctx, TSqlParser::Method_callContext *m
 {
 	std::vector<size_t> keysToRemove;
 	std::string ctx_str = ::getFullText(ctx);
+	ctx_str = ctx_str.substr(0, method->stop->getStopIndex() - ctx->start->getStartIndex() + 1);
 	int func_call_len = (int)geospatial_start_index - ctx->start->getStartIndex();
 	int method_len = (int)method->stop->getStopIndex() - method->start->getStartIndex();
 	std::string expr = "";
 	int index = 0;
 	int offset1 = 0;
-	int offset2 = 0;
 	
 	/* writting the previously rewritten Geospatial context */
 	for (auto &entry : rewritten_query_fragment)
 	{
-		if(entry.first >= ctx->start->getStartIndex() && entry.first <= ctx->stop->getStopIndex())
+		if(entry.first >= ctx->start->getStartIndex() && entry.first <= method->stop->getStopIndex())
 		{
 			expr += ctx_str.substr(index, (int)entry.first - ctx->start->getStartIndex() - index) + entry.second.second;
 			index = (int)entry.first - ctx->start->getStartIndex() + entry.second.first.size();
 			keysToRemove.push_back(entry.first);
 			if(entry.first <= geospatial_start_index) offset1 += (int)entry.second.second.size() - entry.second.first.size();
-			else offset2 += (int)entry.second.second.size() - entry.second.first.size();
 		}
 	}
 	for (const auto &key : keysToRemove) rewritten_query_fragment.erase(key);
 	keysToRemove.clear();
+	expr += ctx_str.substr(index);
 	
-	/* shifting the local id positions to new positions after rewriting the query since they will be quoted later */
+	/* quoting Local ID's here so as to remove possibility of multiple rewrites in a single context */
```

Comment:
Can we have a common function for quoting Local IDs? It is being done at a couple of places.

Reply:
Done